pkg/services/dashboards/dashboard.go

```diff
@@ -3,12 +3,14 @@ package dashboards
 import (
 	"context"
 
+	"github.com/grafana/grafana/pkg/api/dtos"
 	"github.com/grafana/grafana/pkg/models"
 )
 
 //go:generate mockery --name DashboardService --structname FakeDashboardService --inpackage --filename dashboard_service_mock.go
 // DashboardService is a service for operating on dashboards.
 type DashboardService interface {
+	BuildPublicDashboardMetricRequest(ctx context.Context, publicDashboardUid string, panelId int64) (dtos.MetricRequest, error)
```

Comment:
should this be part of this service? could it be the dashboard_public interface? Feels better to have a PublicDashboardService have a dependency on DashboardService than the other way around

Reply:
That's an interesting question... the service is fairly simple right now. I think the more sophisticated sharing config coming in public dashboards v2 would be a reasonable cutpoint to pull out a separate service.

Are there any guidelines around creating a separate service? I admit that I wouldn't want to add another service just because the HTTPServer object is already so crammed.

Reply:
I think it's worth leaving here until we've developed the other pieces that will belong in that service instead of trying to do that now.

Reply:
It feels better to keep things separate , to isolate different concerns and reasons for change (also easier for code ownership)

Reply:
This is being tracked in https://github.com/grafana/grafana-enterprise-partnerships-team/issues/210